### DISCUSSION TITLE
automazione in loop

### DISCUSSION THREAD

**Discussion by fraygon** [indomus/forum#441] [HUB: Home Assistant]
Ciao. Ho una semplice automazione che accende le luci esterne quando il sole tramonta e il sensore di luce rileva buio e mi manda via telegram una info. Funziona ma se da lovelace spengo per qualche motivo il relè di accensione, pur essendoci le condizioni, questo non si riaccende più, devo attivarlo manualmente. Credo sia un problema di lettura in loop dell'automazione ma sinceramente non so fare. Potete suggerirmi come fare? Grazie

**Comment by maxcanna** [comment]
Il punto è che le automazioni scattano al *primo* verificarsi del trigger. Se tu intervieni successivamente in modo manuale l'automazione non lo sa, ha fatto il suo.

Non conosco i dettagli ma ti dovrebbe bastare aggiungere lo stato del relè nel trigger. Così, essendoci di nuovo le condizioni, l'automazione scatta di nuovo.

Presta attenzione a come scrivi il trigger perché altrimenti appena tu spegni la luce questa viene accesa. E non credo che sia quel che vuoi 😉

**Comment by fraygon** [comment]
Ti ringrazio della risposta. Questo è il codice attuale. Amesso di averlo scritto correttamente dove potrei creare il loop o la condizione che mi serve affinchè, anche se spengo per sbaglio la luce, questa si riaccenda se ci sono le due condizioni: sole tramontato e fotocellula superiore a 2.8. Grazie dei suggerimenti
 '''- platform: template
    value_template: '{{ state_attr(''sun.sun'', ''elevation'') < 3 }}'
condition:
  - condition: and
    conditions:
      - condition: sun
        after: sunset
      - condition: numeric_state
        entity_id: sensor.external_brightness
        above: '2.8'
action:
  - service: switch.turn_on
    data: {}
    target:
      entity_id:
        - switch.out_pin_d0_wemos
      device_id: a36d0f6c79c94a0bbe4cb75ca14bd31c
  - service: notify.xxxxxx
    data:
      message: Luci esterne accese
mode: single  '''

**Reply by maxcanna** [reply]
Ci sono una serie di problemi con questa automazione.

- per usare il trigger attuale non hai bisogno di un template, puoi usare l'attributo direttamente 
- non capisco perché hai messo la condizione sullo stato del sole

Quello che ti ho consigliato è di combinare tutti i trigger in un unico template (che puoi provare nei dev tool) che restituisca `True` quando vuoi che scatti. In questo modo l'automazione scatta ogni volta che tutti gli stati sono quelli attesi e non solo la prima volta 

Per usare markdown usa tre ` non uno. Faccio fatica a leggere così

**Reply by fraygon** [reply]
Ho provato come da te suggerito e l'automazione funziona la prima volta ma se spengo io la luce questa non si riaccende. Ho tentato di mettere il trigger anche sullo stato dell'uscita ma non funziona. Naturalmente farò ulteriori tentativi. grazie

**Reply by maxcanna** [reply]
Tieni sott'occhio il template usato come trigger dell'automazione usando i dev tools.
Deve cambiare stato al cambiare dello stato delle entità che hai combinato.
Quando spegni la luce deve ritornare a `True` per far sì che l'automazione scatti di nuovo.

Ma sei sicuro di volere questo? Ti sarebbe impossibile spegnere la luce, ogni volta che la spegni HA te la riaccende

**Reply by fraygon** [reply]
La dovrebbe riaccendere fino all'alba, poi ho un'altra automazione che la spegne.

**Reply by maxcanna** [reply]
Ooook

**Reply by fraygon** [reply]
Ti volevo solo ringraziare per i suggerimenti. Dopo svariati tentativi e letture sui tutorial trovati in rete sono riuscito a fare una automazione decente. Ho provato vari trigger legati al sole ma hanno sempre fallito in qualcosa (a dire il vero non ne ho ancora ben capito il funzionamento in H.A.). Ho quindi creato un trigger chiamato ogni due minuti e lo ho legato alle mie condizioni e sembra tutto ok. Mi è stato molto utile il suggerimento riguardo i dev tools, sapevo cos'erano ma non ne capivo l'effettiva importanza. Magari Domus potrebbe fare un tutorial su questi strumenti molto potenti.

**Reply by maxcanna** [reply]
> Ho quindi creato un trigger chiamato ogni due minuti

non è esattamente quel che ti ho consigliato io ma vabeh, se hai risolto va bene uguale 😅
